include/language-factory.php

```diff
@@ -0,0 +1,278 @@
+	 *     @type string  $w3c             W3C locale.
+	 *     @type string  $flag_code       Code of the flag.
+	 *     @type int     $term_group      Order of the language when displayed in a list of languages.
+	 *     @type int     $is_rtl          `1` if the language is rtl, `0` otherwise.
+	 *     @type int     $mo_id           ID of the post storing strings translations.
+	 *     @type string  $facebook        Optional. Facebook locale.
+	 *     @type string  $home_url        Home URL in this language.
+	 *     @type string  $search_url      Home URL to use in search forms.
+	 *     @type string  $host            Host corresponding to this language.
+	 *     @type string  $flag_url        URL of the flag.
+	 *     @type string  $flag            HTML markup of the flag.
+	 *     @type string  $custom_flag_url Optional. URL of the custom flag if it exists.
+	 *     @type string  $custom_flag     Optional. HTML markup of the custom flag if it exists.
+	 *     @type int     $page_on_front   ID of the page on front in this language.
+	 *     @type int     $page_for_posts  ID of the page for posts in this language.
+	 * }
+	 *
+	 * @phpstan-return array{
+	 *     term_props: array{
+	 *         language: array{
+	 *             term_id: positive-int,
+	 *             term_taxonomy_id: positive-int,
+	 *             count: int<0, max>
+	 *         },
+	 *         term_language: array{
+	 *             term_id: positive-int,
+	 *             term_taxonomy_id: positive-int,
+	 *             count: int<0, max>
+	 *         }
+	 *     },
+	 *     name: non-empty-string,
+	 *     slug: non-empty-string,
+	 *     locale: non-empty-string,
+	 *     w3c: non-empty-string,
+	 *     flag_code: non-empty-string,
+	 *     term_group: int,
+	 *     is_rtl: int<0, 1>,
+	 *     mo_id: positive-int,
+	 *     facebook?: string,
+	 *     home_url: non-empty-string,
+	 *     search_url: non-empty-string,
+	 *     host: non-empty-string,
+	 *     flag_url: non-empty-string,
+	 *     flag: non-empty-string,
+	 *     custom_flag_url?: string,
+	 *     custom_flag?: string,
+	 *     page_on_front:positive-int,
+	 *     page_for_posts:positive-int
+	 * }
+	 */
+	private static function sanitize_data( array $data ) {
+		foreach ( $data['term_props'] as $taxo => $values ) {
+			$values['term_id']             = (int) $values['term_id'];
+			$values['term_taxonomy_id']    = (int) $values['term_taxonomy_id'];
+			$values['count']               = ! empty( $values['count'] ) ? (int) $values['count'] : 0;
+			$data  ['term_props'][ $taxo ] = $values;
+		}
+
+		$string_fields = array( 'name', 'slug', 'locale', 'w3c', 'flag_code', 'facebook', 'home_url', 'search_url', 'host', 'flag_url', 'flag', 'custom_flag_url', 'custom_flag' );
+
+		foreach ( $string_fields as $field ) {
+			if ( empty( $data[ $field ] ) || ! is_string( $data[ $field ] ) ) {
+				$data[ $field ] = '';
+			}
+
+			$data[ $field ] = trim( $data[ $field ] );
+		}
```

Comment:
I don't belive it's useful.

Reply:
I remove it in a14f473
I also add default value to some properties, do you think other properties could benefit from that?